packages/amplify_datastore/lib/amplify_datastore.dart

```diff
@@ -28,16 +28,19 @@ class AmplifyDataStore extends DataStorePluginInterface {
   static final Object _token = Object();
 
   /// Constructs an AmplifyDataStore plugin
-  AmplifyDataStore({@required ModelProviderInterface modelProvider})
+  AmplifyDataStore({required ModelProviderInterface modelProvider})
       : super(token: _token, modelProvider: modelProvider);
 
+  /// Internal use constructor
+  @protected
+  AmplifyDataStore.protected() : super.protected(token: _token);
```

Comment:
For my own edification, could you explain this change?

Reply:
There is a compile issue without this change.  The AmplifyDataStore.protected() will implicitly try to call a constructor with no constructors so we have to explicitly tell it to call the super's protected method with the token parameter.